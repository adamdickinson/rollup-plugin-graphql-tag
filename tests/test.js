const { join } = require('path');
const rollup = require('rollup');
const gql = require('../');

it('should load and transform simple graphql queries', async () => {
  const code = await compile(fixture('simple.js'));
  expect(code).toMatchSnapshot();
});

it('should load and transform graphql schema', async () => {
  const code = await compile(fixture('schema.js'));
  expect(code).toMatchSnapshot();
});

// TODO
// it('should load and transform queries with imports', async () => {
//   const code = await compile(fixture('complex.js'));
//   expect(code).toMatchSnapshot();
// });

// TODO
// it('should load and transform queries as separate exports', async () => {
//   const code = await compile(fixture('multiple.js'));
//   expect(code).toMatchSnapshot();
// });

function fixture(path) {
  return join(__dirname, '__fixtures__', path);
}

async function compile(input) {
  const bundle = await rollup.rollup({
    input,
    plugins: [gql()]
  });

  const { output } = await bundle.generate({
    format: 'es'
  });
  const chunks = output.map(chunk => chunk.code);

  return chunks.length > 1 ? chunks : chunks[0];
}
