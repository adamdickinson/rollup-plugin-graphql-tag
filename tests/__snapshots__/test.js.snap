// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load and transform graphql schema 1`] = `
"var schema = (function() {
  const doc = {\\"kind\\":\\"Document\\",\\"definitions\\":[{\\"kind\\":\\"ObjectTypeDefinition\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"Book\\"},\\"interfaces\\":[],\\"directives\\":[],\\"fields\\":[{\\"kind\\":\\"FieldDefinition\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"title\\"},\\"arguments\\":[],\\"type\\":{\\"kind\\":\\"NamedType\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"String\\"}},\\"directives\\":[]},{\\"kind\\":\\"FieldDefinition\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"author\\"},\\"arguments\\":[],\\"type\\":{\\"kind\\":\\"NamedType\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"Author\\"}},\\"directives\\":[]}]},{\\"kind\\":\\"ObjectTypeDefinition\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"Author\\"},\\"interfaces\\":[],\\"directives\\":[],\\"fields\\":[{\\"kind\\":\\"FieldDefinition\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"name\\"},\\"arguments\\":[],\\"type\\":{\\"kind\\":\\"NamedType\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"String\\"}},\\"directives\\":[]},{\\"kind\\":\\"FieldDefinition\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"books\\"},\\"arguments\\":[],\\"type\\":{\\"kind\\":\\"ListType\\",\\"type\\":{\\"kind\\":\\"NamedType\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"Book\\"}}},\\"directives\\":[]}]}],\\"loc\\":{\\"start\\":0,\\"end\\":103}};
  doc.loc.source = {\\"body\\":\\"\\\\n    type Book {\\\\n  title: String\\\\n  author: Author\\\\n}\\\\n\\\\ntype Author {\\\\n  name: String\\\\n  books: [Book]\\\\n}\\\\n\\\\n  \\",\\"name\\":\\"GraphQL request\\",\\"locationOffset\\":{\\"line\\":1,\\"column\\":1}};

  return doc;
})();

export { schema as typeDefs };
"
`;

exports[`should load and transform simple graphql queries 1`] = `
"var simple = (function() {
  const doc = {\\"kind\\":\\"Document\\",\\"definitions\\":[{\\"kind\\":\\"OperationDefinition\\",\\"operation\\":\\"query\\",\\"variableDefinitions\\":[],\\"directives\\":[],\\"selectionSet\\":{\\"kind\\":\\"SelectionSet\\",\\"selections\\":[{\\"kind\\":\\"Field\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"name\\"},\\"arguments\\":[],\\"directives\\":[]}]}}],\\"loc\\":{\\"start\\":0,\\"end\\":25}};
  doc.loc.source = {\\"body\\":\\"\\\\n    query {\\\\n  name\\\\n}\\\\n\\\\n  \\",\\"name\\":\\"GraphQL request\\",\\"locationOffset\\":{\\"line\\":1,\\"column\\":1}};

  return doc;
})();

var simple$1 = (function() {
  const doc = {\\"kind\\":\\"Document\\",\\"definitions\\":[{\\"kind\\":\\"OperationDefinition\\",\\"operation\\":\\"query\\",\\"variableDefinitions\\":[],\\"directives\\":[],\\"selectionSet\\":{\\"kind\\":\\"SelectionSet\\",\\"selections\\":[{\\"kind\\":\\"Field\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"name\\"},\\"arguments\\":[],\\"directives\\":[]}]}}],\\"loc\\":{\\"start\\":0,\\"end\\":25}};
  doc.loc.source = {\\"body\\":\\"\\\\n    query {\\\\n  name\\\\n}\\\\n\\\\n  \\",\\"name\\":\\"GraphQL request\\",\\"locationOffset\\":{\\"line\\":1,\\"column\\":1}};

  return doc;
})();

export { simple as a, simple$1 as b };
"
`;
